### PR TITLE
Update iframe-modal.ts DOM text reinterpreted as HTML

### PR DESCRIPTION
**Pull request by Shivam7-1**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | By using innerText, it will avoid the risk of HTML injection, as these properties automatically escape any HTML special characters in the provided text. This helps prevent cross-site scripting (XSS) vulnerabilities by treating the input as plain text rather than interpreted HTML.
| Type?             | improvement 
| Category?         |  CO
| BC breaks?        |  no
| Deprecations?     |  no
| How to test?      | CI
| UI Tests          | https://github.com/florine2623/testing_pr/actions/runs/10578640764
| Fixed issue or discussion?     | 
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

**Comment by Shivam7-1** [comment]
Hi @nesrineabdmouleh Could You Please Review This PR 
Thanks

**Comment by nesrineabdmouleh** [comment]
> Hi @nesrineabdmouleh Could You Please Review This PR Thanks

Hello @Shivam7-1, 

Thank you for your first contribution, your fix is not related to UI tests!
@PrestaShop/committers need a review please

**Comment by Shivam7-1** [comment]
Hi @nesrineabdmouleh Thanks You For Responding 
But I think it looks like PrestaShop/committers haven't Got Notified For Review PR ?
Thanks

**Comment by Shivam7-1** [comment]
Hi @nesrineabdmouleh Thanks You For Responding
But I think it looks like @PrestaShop/committers haven't Got Notified For Review This PR ?
Thanks

**Comment by Shivam7-1** [comment]
> > Hi @nesrineabdmouleh Could You Please Review This PR Thanks
> 
> Hello @Shivam7-1,
> 
> Thank you for your first contribution, your fix is not related to UI tests! @PrestaShop/committers need a review please

Hi @nesrineabdmouleh Thanks You For Responding
But I think it looks like @PrestaShop/committers haven't Got Notified For Review This PR ?
Thanks

**Comment by nicosomb** [comment]
@Shivam7-1 Hello. The committers team has been notified, don't worry. 

We'll have a look to your pull request when we'll find time to review it.

**Review by matks — COMMENTED** [review]
Hello @Shivam7-1 apologies for late answer.

Could you fill in the PR template, especially the "How to test"?

Did you verify your code changes do not break any behavior?

**Comment by Shivam7-1** [comment]
Hi @matks Thanks For Your Response 
I had edited PR Template But As I Am New to prestashop I unaware what to write in how to test as it was run locally and also all test was passed in GitHub automatic PR Testing 

Thanks & regards

**Comment by Shivam7-1** [comment]
Hi @matks Could You Please Review This PR 
Thanks

**Comment by kpodemski** [comment]
@matthieu-rolland do you think this PR is relevant?

**Comment by Shivam7-1** [comment]
Hi @matks Could You Please Review This PR
Thanks

**Comment by Shivam7-1** [comment]
Hi @matthieu-rolland Could You Please Review This PR 
Thanks

**Review by matthieu-rolland — COMMENTED** [review]
Using .innerText is indeed safer in general 👍

Could you please write the "how to test" ? 

What should the qa team check to verify that you didn't break anything ?
What page, what action? Where did you spot the xss risk ?

**Comment by Shivam7-1** [comment]
Hi @matthieu-rolland Thanks For Reviewing 
I had updated things in above 
If I am missing Something Let me Know 
Regards

**Comment by Shivam7-1** [comment]
Hi @matks Could You Please Review This PR
Thanks

**Review by boherm — DISMISSED** [review]

**Comment by Shivam7-1** [comment]
Hi @boherm Thanks For Reviewing 
Is there Anything Else is Required From My Side to get this Merged in Project?

Regards

**Comment by Shivam7-1** [comment]
Hi @boherm Thanks For Reviewing
Is there Anything Else is Required From My Side to get this Merged in Project?

Regards

**Comment by boherm** [comment]
> Hi @boherm Thanks For Reviewing Is there Anything Else is Required From My Side to get this Merged in Project?
> 
> Regards

No, just to wait a second approval and an approval by QA team! :)

**Comment by Shivam7-1** [comment]
Hi @matthieu-rolland Could You Please Review This PR 
Thanks

File: admin-dev/themes/new-theme/js/components/modal/iframe-modal.ts
Comment (jolelievre) [inline admin-dev/themes/new-theme/js/components/modal/iframe-modal.ts]: Except the purpose of this render method is also to render HTML, it extends the base method from the `ModalType` and has a few use cases already where it renders HTML code:
https://github.com/PrestaShop/PrestaShop/blob/3a0ae040c5642364f0d1535bd5769dcbdd31f8d5/admin-dev/themes/new-theme/js/pages/product/combination/combination-modal/CombinationModal.vue#L270
https://github.com/PrestaShop/PrestaShop/blob/9541a5198c41a2723abec7d431525e8d030d83d2/admin-dev/themes/new-theme/js/pages/product/category/category-tree-selector.ts#L91

Even if those examples are concrete implementation of a `Modal` and not an `IframeModal` they still share the same base type and so must respect the same contract/behaviour

**Review by jolelievre — CHANGES_REQUESTED** [review]
Sorry @Shivam7-1 

although I understand the purpose of your modification but this component is supposed to be able to render HTML as well

**Comment by Shivam7-1** [comment]
Hi
Thanks For Reviewing is there Any changes is Required 
Thanks

**Comment by jolelievre** [comment]
> Hi Thanks For Reviewing is there Any changes is Required Thanks

I don't think there's much to change, the very principle of your modification is to prevent displaying HTML code, but the feature is made precisely to display HTML code. So your purpose doesn't meet with the expected behaviour

The only alternative I could see would be to add a configuration in the modal, or an additional parameter to the method (less fan) that would indicate if the modal is supposed to protect/prevent HTML code or not

**Comment by Shivam7-1** [comment]
Hi @jolelievre @boherm @matks @matthieu-rolland 
Thanks For Replying According to Feedback I had Tried to make changes 
Could You please Review This Again 

Thanks

**Comment by Shivam7-1** [comment]
Hi 
Is there any Update on This ?
Thanks

**Comment by Shivam7-1** [comment]
Hi @matks 
Is there Anything is Required From My Side ?
Thanks

**Review by jolelievre — DISMISSED** [review]
Thanks @Shivam7-1 

Now the component is more versatile and can handle both use cases

**Comment by Shivam7-1** [comment]
Hi @boherm 
Could You Please Review This PR Again ?
Thanks

**Comment by Shivam7-1** [comment]
Hi @matthieu-rolland 
Could You Please Re-Review This PR Again
Thanks

**Review by kpodemski — APPROVED** [review]

**Review by kpodemski — CHANGES_REQUESTED** [review]
Hello @Shivam7-1 

As you can see, our CI is red because of the errors reported by ESLint:
https://capture.dropbox.com/OgOfcBaye37ELG4f

**Review by matthieu-rolland — COMMENTED** [review]
@Shivam7-1 in CLI, launch this command from the root of your project:

`php vendor/bin/php-cs-fixer fix admin-dev/themes/new-theme/js/components/modal/iframe-modal.ts`

then commit the modified file, it will fix your CI issue 

(you'll need to make a composer install first if your project is not set up already)

**Comment by matthieu-rolland** [comment]
LGTM for me once you've made this last little fix 👍

**Comment by Shivam7-1** [comment]
Hii @matthieu-rolland 
I had Tried this CLI But There is No Change has been Performed BY CLI I had Attached Below Screenshot 
Could You Please Check it 
Screenshot_20240823-104547_1

**Comment by jolelievre** [comment]
@Shivam7-1 the linter failing is for JS code CS fixer is for PHP, sorry

To check the error go to `admin-dev/themes/new-theme` and run `npm run lint` To fix it automatically you can run `npm run lint-fix` but depending on the complexity of the error you may have to fix it manually

**Comment by Shivam7-1** [comment]
Hii @jolelievre @matthieu-rolland 
Still it not Got Run

Could You Please Let me know if there anything I am missing here ?
 
Screenshot_20240823-134826

**Comment by jolelievre** [comment]
@Shivam7-1 did you run `npm ci` or `npm install` before?

**Comment by Shivam7-1** [comment]
Hii @jolelievre  @matthieu-rolland 
Thanks For Replying `npm ci` worked here 
and i had done all ci fix 
Could You Please Again Review This 
Thanks

**Review by jolelievre — APPROVED** [review]

**Comment by Shivam7-1** [comment]
Hi @kpodemski 
Could You Please Review This PR Again?
Thanks

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
@PrestaShop/qa-functional I think this change doesn't require functional tests and CI is ok

**Comment by Shivam7-1** [comment]
Hii
Is there Anything Else is Required From My Side to get this PR merge?
Thanks

**Comment by kpodemski** [comment]
@Shivam7-1 yes: be patient :D

**Comment by Shivam7-1** [comment]
Hii
Is there Anything Else is Required From My Side to get this PR merge?
Thanks :D

**Review by florine2623 — APPROVED** [review]
Hello @Shivam7-1 and @kpodemski ,

Automated tests are OK ✅ 

It can be merged.

**Pull request merged**
